### PR TITLE
Final take, not including bonus

### PR DESCRIPTION
**Pull request by deeprave**
Not entirely sure how to convert the detail views, but will take a
shot at it later and add it to this PR when working.

### CODE REVIEW

File: timetracker/entries/forms.py [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-	name = forms.CharField()
-	client = forms.ModelChoiceField(queryset=Client.objects.all())
+class ProjectForm(forms.ModelForm):
+    class Meta:
+        model = Project
+        fields = ('client','name')
 
 
-class EntryForm(forms.Form):
-    start = forms.DateTimeField(label="Start Time", help_text="Format: 2006-10-25 14:30")
```
Comment (MarkusH) [inline timetracker/entries/forms.py:17]: You're loosing the help text for the form fields here. Either define them as `help_texts` on `EntryForm.Meta` or as part of the respective model field.

File: timetracker/entries/urls.py [hunk 2]
```diff
@@ -4,9 +4,10 @@
 
 
 urlpatterns = [
-    url(r'^clients/$', views.clients, name='client-list'),
+	url(r'^$', views.RootRedirectView.as_view(), name='root'),
```
Comment (MarkusH) [inline timetracker/entries/urls.py]: You should be careful with using tabs and spaces for indentation. Recommendation is to use 4 spaces.

File: timetracker/entries/views.py [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
+    template_name = 'clients.html'
+    form_class = ClientForm
 
-    client_list = Client.objects.all()
-    return render(request, 'clients.html', {
-        'client_list': client_list,
-        'form': form,
-    })
+    def get_success_url(self):
+        return reverse('client-list')
```
Comment (MarkusH) [inline timetracker/entries/views.py]: You can use `success_url = reverse_lazy('client-list')` instead :wink: (`reverse_lazy` is at the same place as `reverse`)

File: timetracker/entries/views.py [hunk 4]
```diff
@@ -111,3 +88,10 @@ def project_detail(request, pk):
         'project': project,
         'form': form,
     })
+
+
+class RootRedirectView(RedirectView):
+    permanent = False
+
+    def get_redirect_url(self):
```
Comment (MarkusH) [inline timetracker/entries/views.py]: You can use `redirect_url = reverse_lazy('client-list')` instead :wink: (`reverse_lazy` is at the same place as `reverse`)

**Comment by MarkusH** [comment]
Looks good so far :smile:

File: timetracker/entries/forms.py (hunk 1 above)
Comment (deeprave) [inline timetracker/entries/forms.py:17]: Whoops, fixed in coming update. Thanks!

File: timetracker/entries/urls.py (hunk 2 above)
Comment (deeprave) [inline timetracker/entries/urls.py]: Some weirdness with sublime text going on here. I caught it doing this once but missed this one. It is set to edit python code with no hard tabs. (I normally use PyCharm which to my knowledge never gets this wrong)

**Pull request closed without merging**